torch/distributed/nn/jit/code_template.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+
+import re
+
+# match $identifier or ${identifier} and replace with value in env
+# If this identifier is at the beginning of whitespace on a line
+# and its value is a list then it is treated as
+# block subsitution by indenting to that depth and putting each element
+# of the list on its own line
+# if the identifier is on a line starting with non-whitespace and a list
+# then it is comma separated ${,foo} will insert a comma before the list
+# if this list is not empty and ${foo,} will insert one after.
```

Comment:
Why do we need to build this ourselves, can't we just use python's format(https://docs.python.org/3.4/library/functions.html#format) to appropriately replace the correct env variables?

Reply:
Replace if with python `str.format(..)`.